### PR TITLE
refactor suite workflow class and improve error message

### PR DESCRIPTION
**Pull request by snopoke**
http://manage.dimagi.com/default.asp?183479
@dannyroberts 

easiest review by commit 

buddy @millerdev

### CODE REVIEW

File: corehq/apps/app_manager/suite_xml/post_process/workflow.py [hunk 1]
```diff
@@ -26,147 +26,143 @@ def __init__(self, suite, app, modules):
...
                 stack_frames = []
-                case_list_form_frames = self.case_list_forms_frames(form)
+                case_list_form_frames = CaseListFormWorkflow(self).case_list_forms_frames(form)
                 stack_frames.extend(case_list_form_frames)
 
                 if case_list_form_frames:
                     if_prefix = session_var(RETURN_TO).count().eq(0)
 
-                stack_frames.extend(self.form_workflow_frames(if_prefix, module, form))
+                stack_frames.extend(EndOfFormNavigationWorkflow(self).form_workflow_frames(if_prefix, module, form))
```
Comment (dimagimon) [inline corehq/apps/app_manager/suite_xml/post_process/workflow.py]: E501 line too long (116 > 115 characters)

**Comment by snopoke** [comment]
@millerdev updated

**Comment by dannyroberts** [comment]
Looks like this improves the unexpected error text, but doesn't prevent that error from happening with build validation. Is that right?

**Comment by dannyroberts** [comment]
If so, looks good!

**Comment by snopoke** [comment]
@dannyroberts some additional validation in this PR: https://github.com/dimagi/commcare-hq/pull/8569

**Comment by millerdev** [comment]
:+1: feel free to merge after tests pass

**Pull request merged**